src/Compute/v2/Models/Server.php

```diff
@@ -314,6 +314,19 @@ public function listAddresses(array $options = []): array
     }
 
     /**
+     * Iterates over all the ports for this server.
+     *
+     * @return array
+     */
+    public function listPorts(array $options = []): array
+    {
+        $options['id'] = $this->id;
+
+        $response = $this->execute($this->api->getPorts(), ['id' => $this->id]);
```

Comment:
Instead of `array`, I think it is better to encapsulate in class `InterfaceAttachment`. See [VolumeAttachment](https://github.com/php-opencloud/openstack/blob/master/src/BlockStorage/v2/Models/VolumeAttachment.php) and [Server:: listVolumeAttachments](https://github.com/php-opencloud/openstack/blob/master/src/Compute/v2/Models/Server.php#L434)